src/Oliver.Common/Infrastructure/FileLogger.cs

```diff
@@ -28,18 +28,21 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             {
                 if (!Directory.Exists(Path.GetDirectoryName(this.fileName)))
                     Directory.CreateDirectory(Path.GetDirectoryName(this.fileName));
+            }
 
-                var msgs = new List<string>
+            var msgs = new List<string>
                 {
                     $"{DateTime.Now.ToString("yyyy.MM.dd HH:mm:ss")}|{logLevel.ToString()}|{this.categoryName}| {formatter(state, exception)}",
                 };
 
-                if (exception != null)
-                {
-                    msgs.Add($"{DateTime.Now.ToString("yyyy.MM.dd HH:mm:ss")}|{logLevel.ToString()}|{this.categoryName}| {exception?.Message}");
-                    msgs.Add($"{DateTime.Now.ToString("yyyy.MM.dd HH:mm:ss")}|{logLevel.ToString()}|{this.categoryName}| {exception?.StackTrace}");
-                }
+            if (exception != null)
+            {
+                msgs.Add($"{DateTime.Now.ToString("yyyy.MM.dd HH:mm:ss")}|{logLevel.ToString()}|{this.categoryName}| {exception?.Message}");
+                msgs.Add($"{DateTime.Now.ToString("yyyy.MM.dd HH:mm:ss")}|{logLevel.ToString()}|{this.categoryName}| {exception?.StackTrace}");
+            }
 
+            lock (this.locker)
```

Comment:
This `locker` should be unique for the specific file, but now if we create 2 logger instances with the same file name (and yes, **public** `FileLoggerProvider.CreateLogger`) allows us to do that), this check will fail

Reply:
moved to private nested class under FileLoggerProvider to avoid this situation

Reply:
It hasn't resolved the issue cause we still can create 2 or more loggers over the same file with different locks.